app/(root)/profile/[id]/page.tsx

```diff
@@ -1,20 +1,36 @@
 import ProfileHeader from "@/components/shared/profile/ProfileHeader";
-
-import { getOneUser } from "@/lib/actions/user.actions";
+import TechStacks from "@/components/shared/profile/TechStacks";
+import { getOneUser } from "@/lib/actions/user.actions"; // Removed unused import
 import { getSession } from "@/lib/authOptions";
+import ExperienceLevels from "@/components/shared/profile/ExperienceLevels";
 import React from "react";
-// profile of the user will go here
+import LearningGoals from "@/components/shared/profile/LearningGoals";
+
+// TODO: Typing and styling for each component
 
 const ProfilePage = async () => {
   const session = await getSession();
+  let user;
 
-  const user = await getOneUser(session?.user?.email!);
-  if (user) {
-    console.log(user.username);
+  if (session) {
+    user = await getOneUser(session?.user?.email!);
   }
+
   return (
-    <div className="flex flex-auto flex-col">
-      {user && <ProfileHeader user={user} />}
+    <div className="box-border flex min-h-[screen] w-full flex-col ">
+      {user && <ProfileHeader />}
+      {user && (
+        <LearningGoals user={user} learningGoals={user.learningGoals ?? []} />
+      )}
+      {user && (
+        <ExperienceLevels
+          user={user}
+          experienceLevels={user.experienceLevel ?? []}
+        />
+      )}
+      {user && (
+        <TechStacks user={user} technologies={user.technologyStack ?? []} />
+      )}
```

Comment:
Wouldn't it make more sense to return something else if there is no user above this instead of `user &&` 4 times?

Above the return:
if(!user){
    return "No User" // or redirect, or throw an error or 404
}

Then you know there's a user for the rest of the return.